com.unity.perception/Runtime/GroundTruth/Labelers/BoundingBoxLabeler.cs

```diff
@@ -108,8 +116,73 @@ void OnRenderedObjectInfosCalculated(int frameCount, NativeArray<RenderedObjectI
                     };
                 }
 
+                if (!CaptureOptions.useAsyncReadbackIfSupported && frameCount != Time.frameCount) 
+                    Debug.LogWarning("Not on current frame: " + frameCount + "(" + Time.frameCount + ")");
+
+                if (IsVisualizationEnabled()) 
+                {
+                    Visualize();
+                    ReportFrameCount(frameCount);
+                }
+                
                 asyncAnnotation.ReportValues(m_BoundingBoxValues);
             }
         }
+
+        /// <inheritdoc/>
+        protected override void SetupVisualizationPanel(GameObject panel)
+        {
+            var toggle  = GameObject.Instantiate(Resources.Load<GameObject>("GenericToggle"));
```

Comment:
Thinking about the experience from a user perspective - will a user be expected to use Resources.Load() to instantiate UI prefabs? I'm a bit wary of asking users to depend on the presense and internals of specific prefabs in the package. Maybe we could wrap this with a little factory class and then give users the freedom to add their own UI by hand?

Reply:
So, are you suggesting to get rid of prefabs, and just create the genericToggle and generic slider on the fly? If so, that would allow us to get rid of resources all together. Unfortunately, I'll have to fight a little bit with rect transform layouts, which was a big pain, and the overall reason why I went the prefab direction to begin with.

Reply:
It it makes sense to do so, yes. At least what I'm saying is that users should not have to write `GameObject.Instantiate(Resources.Load<GameObject>("GenericToggle"))` . Instead, they should be able to do something like `Visualization.CreateToggle()` or even just `AddToggle()`

Reply:
This has been fixed as a part of my UI rearchitecture